spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,80 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice. The value of the property can contain only characters that are valid for config property names. 
+This is because the name of the profile is directly stored in the name of the config property. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The property `mp.config.profile` is only read once on microservice starting. Any changes after that is ignored. Furthermore, the value specifies a single profile. Only one profile can be active at a time.
+Commas in the value have no special meaning. For example, if the value of `mp.config.profile` is `testing,live`,  a single profile named `testing,live` is enabled instead of enabling two profiles. 
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile.`, (e.g. `%live.vehicle.name`).
+
+For an instance, a config source can be specified as follows.
+
+[source, text]
+----
+%dev.vehicle.name=car
+%live.vehicle.name=train
+%testing.vehicle.name=bike
+vehicle.name=lorry
+----
+
+A config property associated with the Config Profile can be looked up as follows.
+
+[source, text]
+----
+@Inject @ConfigProperty(name="vehicle.name") String vehicleName;
+----
+
+[source, text]
+----
+String vehicleName = ConfigProvider.getConfig().getValue("vehicle.name", String.class);
+----
+
+If the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` is the Active Property. An active property overrides the properties in the same config source. 
+In more details, if the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` overrides the property `vehicle.name`. 
+The properties `%live.vehicle.name` and `%testing.vehicle.name` are inactive config property and don't override the property `vehicle.name`.
+
+In more details, on setting the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` is the active property. The `vehicleName` is `train`.
+On setting the system property `mp.config.profile` is set to `live`, the property `%live.vehicle.name` is the active property. The `vehicleName` is `train`. Similarly, `bike` will be the value of `vehicleName`, if the profile is `testing`.
+
+When retrieving the config values, this Config spec implementations must first search for the value of `mp.config.profile` and then retrieve the value of the config property `%${mp.config.profile}.vehicle.name`,
```

Comment:
I don't think that using the proposed property expansion syntax here is a good idea, since that proposal has not yet been accepted; I think it would just be confusing.  Also, I don't think this counts as "normative" since it's really an example.

I would add some wording like:

> Conforming implementations are required to return the value of the "profile-specific" property name whenever the "profile-specific" property name occurs within a configuration source whose overall priority¹ is equal to, or higher than, the value corresponding to the original property name (if any).  A "profile-specific" property name is defined as a property whose name consists of the following sequence: `%` \<profile name> `.` \<original property name>

Add the footnote for "overall priority" because configuration source ordering is more complex than just ordinal value.  That definition should be specified & linked from here.

Reply:
I removed the property expansion syntax but reworded slightly. Please double check to see whether it is understandable.